pkg/auth/config/config.go

```diff
@@ -1,47 +1,108 @@
+	// will not be issued using this key.
+	SecretOldTokenSigningRSAKey = "token_rsa_key_old.pem"
+)
 
+//go:generate enumer --type=AuthorizationServerType --trimprefix=AuthorizationServerType -json
+
+// AuthorizationServerType defines the type of Authorization Server to use.
+type AuthorizationServerType uint8
+
+const (
+	// AuthorizationServerTypeSelf determines that FlyteAdmin should act as the authorization server to serve
+	// OAuth2 token requests
+	AuthorizationServerTypeSelf AuthorizationServerType = iota
+
+	// AuthorizationServerTypeExternal determines that FlyteAdmin should rely on an external authorization server (e.g.
+	// Okta) to serve OAuth2 token requests
+	AuthorizationServerTypeExternal
+)
+
+var (
+	defaultConfig = &Config{
+		// Please see the comments in this struct's definition for more information
+		HTTPAuthorizationHeader: "flyte-authorization",
+		GrpcAuthorizationHeader: "flyte-authorization",
+		UserAuth: UserAuthConfig{
+			CookieHashKeySecretName:  SecretCookieHashKey,
+			CookieBlockKeySecretName: SecretCookieBlockKey,
+			OpenID: OpenIDOptions{
+				ClientSecretName: SecretOIdCClientSecret,
+				// Default claims that should be supported by any OIdC server. Refer to https://openid.net/specs/openid-connect-core-1_0.html#ScopeClaims
+				// for a complete list.
+				Scopes: []string{
+					"openid",
+					"profile",
+				},
+			},
+		},
+		AppAuth: OAuth2Options{
+			AuthServerType: AuthorizationServerTypeSelf,
+			SelfAuthServer: AuthorizationServer{
+				AccessTokenLifespan:             config.Duration{Duration: 30 * time.Minute},
+				RefreshTokenLifespan:            config.Duration{Duration: 60 * time.Minute},
+				AuthorizationCodeLifespan:       config.Duration{Duration: 5 * time.Minute},
+				TokenHashKeySecretName:          SecretTokenHash,
+				TokenSigningRSAKeySecretName:    SecretTokenSigningRSAKey,
+				OldTokenSigningRSAKeySecretName: SecretOldTokenSigningRSAKey,
+				StaticClients: map[string]*fosite.DefaultClient{
+					"flyte-cli": {
+						ID:            "flyte-cli",
+						RedirectURIs:  []string{"http://localhost:53593/callback", "http://localhost:12345/callback"},
+						ResponseTypes: []string{"code", "token"},
+						GrantTypes:    []string{"refresh_token", "authorization_code"},
+						Scopes:        []string{"all", "offline", "access_token"},
+						Public:        true,
+					},
+					"flytectl": {
+						ID:            "flytectl",
+						RedirectURIs:  []string{"http://localhost:53593/callback", "http://localhost:12345/callback"},
+						ResponseTypes: []string{"code", "token"},
+						GrantTypes:    []string{"refresh_token", "authorization_code"},
+						Scopes:        []string{"all", "offline", "access_token"},
+						Public:        true,
+					},
+					"flytepropeller": {
+						ID:            "flytepropeller",
+						Secret:        []byte(`$2a$10$IxMdI6d.LIRZPpSfEwNoeu4rY3FhDREsxFJXikcgdRRAStxUlsuEO`), // = "foobar"
```

Comment:
to flytepropeller for example? yes, I'm going to add it to the sandbox deployment so that flytepropeller has it already deployed... and then in the guide we should tell people how to change it...